AnnoDesigner/Localization/TranslateExtension.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace AnnoDesigner.Localization
+{
+    public class TranslateKeyConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var translations = value as IDictionary<string, string>;
+
+            if (parameter is string key && translations.TryGetValue(key, out var translation))
+            {
+                return translation;
+            }
+#if DEBUG
+            throw new Exception($"Missing translation or not string translation key \"{parameter}\"");
+#else
+            return $"!{parameter}";
+#endif
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class Localize : Binding
+    {
+        private static TranslateKeyConverter TranslateConverter { get; } = new TranslateKeyConverter();
```

Comment:
Property should be renamed to `LocalizeConverter`, which will match once the `TranslateKeyConverter` class has been renamed above.